problem4.js

```diff
@@ -0,0 +1,212 @@
+    if(historyItem.money>0){
+      AllInput += historyItem.money;
+    }
+    else{
+      AllOutput += Math.abs(historyItem.money);
+    }
+//                        click관련된 함수 
+    deleteButton.onclick=function(){
+      detailsSection.removeChild(itemElement);
+      INIT_BALANCE -= historyItem.money;
+      balanceElement.textContent = INIT_BALANCE;
+      
+      if(historyItem.money>0){
+        AllInput -= historyItem.money;
+        AllInputadd.textContent ="총 수입 "+AllInput;
+      }
+      else{
+        AllOutput -= Math.abs(historyItem.money);
+        AllOutputadd.textContent ="총 지출 "+AllOutput;
+      }
+      
+    };
+
+    itemElement.appendChild(categoryElement);
+    itemElement.appendChild(detailElement);
+    itemElement.appendChild(moneyElement);
+    itemElement.appendChild(deleteButton);
+    detailsSection.appendChild(itemElement);
+  }
+  balanceElement.textContent = INIT_BALANCE;
+  AllInputadd.textContent ="총 수입 "+AllInput;
+  AllOutputadd.textContent ="총 지출 "+AllOutput;
+
+
+}
+Showlist(HISTORY_LIST);
+
+const inputBox = document.getElementById("input_box");
+const outputBox = document.getElementById("output_box");
+
+function filtering() {
+  if (inputBox.checked && outputBox.checked) {
+    Showlist(HISTORY_LIST);
+  } else if (inputBox.checked) {
+    Showlist(HISTORY_income);
+  } else if (outputBox.checked) {
+    Showlist(HISTORY_spend);
+  } else {
+    Showlist(emptylist);
+  }
+}
+
+inputBox.addEventListener("change", filtering);
+outputBox.addEventListener("change", filtering);
+
+
+var modal = document.getElementById('entryModal');
+var showModalBtn = document.getElementById('showModalBtn');
+var closeBtn = document.getElementsByClassName('close')[0];
+var addEntryBtn = document.getElementById('addEntry');
+var incomeBtn = document.getElementById('incomeBtn');
+var expenseBtn = document.getElementById('expenseBtn');
+var typeSelect = document.getElementById('typeSelect');
+var amountInput = document.getElementById('amountInput');
+var descriptionInput = document.getElementById('descriptionInput');
+
+var currentType = '';  // '수입' 또는 '지출'
+
+showModalBtn.addEventListener('click', function() {
+  modal.style.display = 'block';
+});
+
+incomeBtn.addEventListener('click', function() {
+  currentType = '수입';
+});
+
+expenseBtn.addEventListener('click', function() {
+  currentType = '지출';
+});
+
+closeBtn.addEventListener('click', function() {
+  modal.style.display = 'none';
+});
+
+// 모달 바깥쪽 클릭 시 모달 닫기
+window.addEventListener('click', function(event) {
+  if (event.target == modal) {
+    modal.style.display = 'none';
+  }
+});
+
+
+
+addEntryBtn.addEventListener('click', function() {
+  // 수입/지출, 종류, 금액, 내용 값을 가져옵니다.
+  const type = currentType;
+  const category = typeSelect.value;
+  const detail = descriptionInput.value;
+  let money = parseFloat(amountInput.value);
+  if((money == 0)||isNaN(money)){
+    return;
+  
+  };
+
+  // 지출일 경우 음수로 변환
+  if (type === '지출') {
+    money = -money;
+  }
```

Comment:
될 수 있으면 삼항 이항 연산자를 추천합니다!
```suggestion
  type === '지출' &&    money = -money;
  
```